cpp/ql/src/Critical/GlobalUseBeforeInit.ql

```diff
@@ -98,10 +98,25 @@ predicate callReaches(Call call, ControlFlowNode successor) {
   )
 }
 
+// To avoid many false alarms like `static int a = 1;`
+predicate initialisedAtDeclaration(GlobalVariable v) {
+  exists(VariableDeclarationEntry vde |
+    vde = v.getDefinition()
+    and vde.isDefinition()
+  )
+}
+
+// No need to initialize those variables
+predicate isStdlibVariable(GlobalVariable v) {
+  v.getName() = ["stdin", "stdout", "stderr"]
+}
+
 from GlobalVariable v, Function f
 where
   uninitialisedBefore(v, f) and
-  useFunc(v, f)
+  useFunc(v, f) and
+  not initialisedAtDeclaration(v) and
+  not isStdlibVariable(v)
```

Comment:
_Ideally_, these two new conditions should be put into the base case of the recursion in `uninitialisedBefore` for performance reasons. But I don't think it matters a lot since the query already performs fine as-is, so feel free to leave that optimization out 🙂

Reply:
Fixed in https://github.com/github/codeql/pull/16149/commits/439afd97baff457b0e6624e1fc3ae5ba8f565cce